File: 0509-Fibonacci-Number/0509.java
class Solution {
    public int fib(int n) {
        if (n <= 1)
            return n;
        int x = 0, y = 1, ans = 0;

        for (int i = 0; i < n - 1; i++) {
            ans = x + y;
            x = y;
            y = ans;
        }

        return ans;
    }
}

File: 0509-Fibonacci-Number/0509.py
class Solution:
    def fib(self, n: int) -> int:
        if n <= 1: return n
        # x = first preceding number, y = second preceding number
        x, y = 0, 1
        ans = 0

        for i in range(n - 1):
            ans = x + y
            x = y
            y = ans
        
        return ans
        